src/services/registry.js

```diff
@@ -83,13 +85,13 @@ class RegistryService {
     }
 
     try {
-      await token.approve(this.address, deposit)
+      await token.approve(this.address, bigDeposit)
     } catch (error) {
       throw error
     }
 
     try {
-      await this.registry.apply(domain, deposit, {from: this.getAccount()})
+      await this.registry.apply(domain, bigDeposit, {from: this.getAccount()})
```

Comment:
does it work without calling `<BN>.toNumber()` when passing as argument? @kangarang

Reply:
@miguelmota it's supported, but we should always use `BigNumber`s for precision.

https://hackernoon.com/a-note-on-numbers-in-ethereum-and-javascript-3e6ac3b2fad9

Reply:
```
Invalid number of arguments to Solidity function
```

Seems to be getting it confused with the options object. Works when using `toNumber()`

Reply:
i see, yes, i'm getting the same error now. will report back soon

Reply:
Try `toString()`. Should work, and safer than `toNumber()`.

Reply:
Truffle-contract uses BigNumber, not BN, and they're usually interoperable but going to a string is pretty foolproof, and also lossless.

Reply:
AHH that makes sense! thanks @skmgoldin